File: app/src/test/java/com/orangepenguin/boilerplate/util/SharedPreferenceUtilInMemoryImpl.java
package com.orangepenguin.boilerplate.util;

import java.util.HashMap;
import java.util.Map;

public class SharedPreferenceUtilInMemoryImpl implements SharedPreferencesUtil {

    private static Map<String, String> repository = new HashMap<>();

    @Override
    public String getPreference(String key, String defaultValue) {
        if (!repository.containsKey(key)) {
            return defaultValue;
        }
        return repository.get(key);
    }

    @Override
    public void savePreference(String key, String value) {
        repository.put(key, value);
    }

    @Override
    public void clearPreference(String key) {
        repository.remove(key);
    }
}

File: app/src/test/java/com/orangepenguin/boilerplate/util/SharedPreferenceUtilInMemoryImpl.kt
package com.orangepenguin.boilerplate.util

class SharedPreferenceUtilInMemoryImpl : SharedPreferencesUtil {

    override fun getPreference(key: String, defaultValue: String?): String? =
            stringRepository[key] ?: defaultValue

    override fun getPreference(key: String, defaultValue: Boolean): Boolean =
            booleanRepository[key] ?: defaultValue

    override fun savePreference(key: String, value: String) {
        stringRepository[key] = value
    }

    override fun savePreference(key: String, value: Boolean) {
        booleanRepository[key] = value
    }

    override fun clearPreference(key: String) {
        stringRepository.remove(key)
    }

    companion object {
        private val stringRepository = HashMap<String, String>()
        private val booleanRepository = HashMap<String, Boolean>()
    }
}
